### PR TITLE
collect the metric node_reboot_required

### PR DESCRIPTION
**Pull request by Rabie-Zamane**
Render the metric node_reboot_required If yum-utils is installed.

### CODE REVIEW

File: yum.sh [hunk 1]
```diff
@@ -37,3 +37,15 @@ if [[ -n "${upgrades}" ]] ; then
 else
   echo 'yum_upgrades_pending{origin=""} 0'
 fi
+
+# if yum-utils is not installed the skip rendering this metric 
+if [[ -f '/bin/needs-restarting' ]] ; then
```
Comment (dswarbrick) [inline yum.sh:42]: This ought to be `-x` instead of `-f`, since we need the file to exist _and_ be executable.

File: yum.sh [hunk 2]
```diff
@@ -37,3 +37,15 @@ if [[ -n "${upgrades}" ]] ; then
...
+  echo '# HELP node_reboot_required Node reboot is required for software updates.'
+  echo '# TYPE node_reboot_required gauge'
+  /bin/needs-restarting -r  > /dev/null 2>&1
+  if [[ $? -eq 0 ]] ; then
```
Comment (dswarbrick) [inline yum.sh:46]: The exit code of `needs-restarting` can be checked directly. See https://github.com/koalaman/shellcheck/wiki/SC2181 for an example.

**Comment by SuperQ** [comment]
The commits also need to have a DCO sign-off. You can generally use `git commit -s --amend`.

**Comment by dswarbrick** [comment]
@Rabie-Zamane Last call for fixing the outstanding issues on this PR. Gates are closing...

**Comment by dswarbrick** [comment]
Closing in favour of #165.

**Pull request closed without merging**